lib/IRGen/GenCall.cpp

```diff
@@ -1117,6 +1129,9 @@ void SignatureExpansion::expandExternalSignatureTypes() {
   // Convert each parameter to a Clang type.
   for (auto param : params) {
     auto clangTy = IGM.getClangType(param, FnType);
+    if (clangTy->isVoidType()) {
+      continue;
+    }
```

Comment:
This probably deserves a comment explaining that you're smuggling ignored parameters through lowering this way.

Reply:
An alternative (maybe better?) approach would be skipping metatypes two lines above before we ever even convert them to void. Then we could remove support for thin metatypes in `GenClangType` too.

Reply:
Why not just drop this argument from the SILFunctionType instead of working around it in IRGen?  I'm positive we have support for that in SIL type lowering; maybe it needs to be extended to constructors, but that shouldn't be hard.

Reply:
Yeah, I just thinking about the possibility of doing this. I wonder how many things depend on the constructor taking a metatype. I bet you're right, we could just remove this parameter from the SILFunction. It's going to get removed at some point anyway.

Reply:
I've update this patch to skip the self metatype parameter. Let me know what you think.